hibike/hibike_message.py

```diff
@@ -1,55 +1,57 @@
 from __future__ import print_function
 # Rewritten because Python.__version__ != 3
-import serial
 import struct
 import pdb
-# Dictionary of message types: message id
-messageTypes = {
-  "SubscriptionRequest" :  0x00,
-  "SubscriptionResponse" : 0x01,
-  "DataUpdate" :           0x02,
-  "DeviceUpdate" :         0x03,
-  "DeviceStatus" :         0x04,
-  "DeviceResponse" :       0x05,
-  "Ping" :                 0x06,
-  "DescriptionRequest" :   0x08,
-  "DescriptionResponse" :  0x09,
-  "Error" :                0xff
+import os
+import json
+
+config_file = open(os.path.join(os.path.dirname(__file__), 'hibikeDevices.json'), 'r')
+devices = json.load(config_file)
+
+paramMap = {device["id"]: {param["name"]: (param["number"], param["type"], param["read"], param["write"]) for param in device["params"]} for device in devices}
+devices = {device["id"]: device for device in devices}
+"""
+structure of devices
+{0: 
+    {"id": 0, "name": "LimitSwitch", "params": [
+                                               {"number": 0, "name": "switch0", "type": "bool", "read": True, "write": False}
+                                               ]
+    }
 }
 
-# Dictionary of device types: enumeration
-deviceTypes = {
-  "Limit Switch" :         0x00,
-  "Line Follower" :        0x01,
-  "Potentiometer" :        0x02,
-  "Encoder" :              0x03,
-  "Battery Buzzer" :       0x04,
-  "Team Flag" :            0x05,
-  "Grizzly" :              0x06,
-  "Servo Control" :        0x07,
-  "Linear Actuator" :      0x08
+"""
+
+# Dictionary mapping param types to python struct format characters
+paramTypes = {
+  "bool" : "?",
+  "uint8_t": "B",
+  "int8_t": "b", 
+  "uint16_t": "H",
+  "int16_t": "h", 
+  "uint32_t": "I",
+  "int32_t": "i", 
+  "uint64_t": "Q",
+  "int64_t": "q", 
+  "float": "f",
+  "double": "d"
 }
 
-
-# Dictionary of message types: payload length
-messagePayloadLengths = {
-  messageTypes["SubscriptionRequest"] :  2,
-  messageTypes["SubscriptionResponse"] : 13,
-  # TODO: find a clean way to not hardcode this
-  messageTypes["DataUpdate"] :           1,
-  messageTypes["DeviceUpdate"] :         5,
-  messageTypes["DeviceStatus"] :         5,
-  messageTypes["DeviceResponse"] :       5,
-  messageTypes["Error"] :                1
+# Dictionary of message types: message id
+messageTypes = {
+  "Ping" :                 0x10,
```

Comment:
As I mentioned last PR, the move to python3 has given us access to an Enum class that can better be used for `messageTypes` and `errorCodes`. That said, I would be fine adding those to the issue tracker rather than addressing in this PR.